src/array.cr

```diff
@@ -1549,6 +1549,17 @@ class Array(T)
     dup.sort!
   end
 
+  # Returns an array with all elements in the collection sorted based on the return value of the block given.
+  #
+  #```
+  # a = [3, 1, 2]
+  # b = a.sort { |a,b| b <=> a }
+  #
+  # b # => [3, 2, 1]
+  # a # => [3, 1, 2]
+  # ```
+  # The block given must implement a comparison, either with the comparison operator `<=>`
+  # or a comparison between *a* and *b*, where a < b yields -1, a == b yields 0, and a > b yields 1.
```

Comment:
yields => returns (reduces confusion)

Reply:
I'd suggest to explain the return values first and then mention the `<=>` operator:

```
The given block must implement a comparison between two elements *a* and *b*,
where `a < b` returns `-1`, `a == b` returns `0`, and `a > b` returns `1`.
The comparison operator `<=>` can be used for this.
```

This should go before the example because the example code.

Reply:
@straight-shoota - This was lifted from `Array#sort`:
https://github.com/marksiemers/crystal/blob/689ca4df14540a20823395dfac1893ca6c7b3d98/src/array.cr#L1546

I'm happy to change it here. Should it be removed or modified in `Array#sort` also?

Reply:
Yeah that needs to be fixed as well. The method description for `#sort` without a block doesn't need this.